lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -106,6 +113,22 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
     }
   }
 
+  void killAfterTimeout() {
+      ::sleep(10);
+
+      if (!done) {
```

Comment:
I'm joining the thread now, but that means we'll always wait those 10 seconds. Do you think that is fine or should we make the sleep interruptible?